### PR TITLE
Bump workflow-dispatch to v4.0.0

### PR DESCRIPTION
**Pull request by ichengchang**
Ticket: [DDO-3037](https://github.com/broadinstitute/workflow-dispatch/pull/7)
<Put notes here to help reviewer understand this PR>

Update all `workflow-dispatch@v3` to `workflow-dispatch@v4.0.0`

There is an associated [PR](https://github.com/broadinstitute/terra-github-workflows/pull/117) with addition of `run-name` to various workflow files.

This enables workflow dispatch to include `run-name` as an input parameter to resolve false positives / negatives amidst concurrent workflows.

See this [blog](https://broadworkbench.atlassian.net/wiki/spaces/IRT/pages/2826928129/Race+Conditions+in+Asynchronous+GitHub+Action+Workflows) post for details.

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

[DDO-3037]: https://broadworkbench.atlassian.net/browse/DDO-3037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ichengchang** [comment]
jenkins retest

**Comment by ichengchang** [comment]
jenkins retest

File: .github/workflows/consumer_contract_tests.yml [hunk 1]
```diff
@@ -119,27 +123,43 @@ jobs:
...
     steps:
       - name: Dispatch to terra-github-workflows
-        uses: broadinstitute/workflow-dispatch@v3
+        uses: broadinstitute/workflow-dispatch@v4.0.0
         with:
+          run-name: "${{ env.PUBLISH_CONTRACTS_RUN_NAME }}"
           workflow: .github/workflows/publish-contracts.yaml
           repo: broadinstitute/terra-github-workflows
-          ref: refs/heads/main
+          ref: refs/heads/iv-upgrade-workflow-dispatch-to-support-run-name
```
Comment (cahrens) [inline .github/workflows/consumer_contract_tests.yml]: Remember to go through and update all of these after the other PR merges.

**Review by cahrens — APPROVED** [review]
LGTM! I'm very excited to have the false failures fixed-- thank you!

File: .github/workflows/consumer_contract_tests.yml (hunk 1 above)
Comment (ichengchang) [reply .github/workflows/consumer_contract_tests.yml]: The other [PR](https://github.com/broadinstitute/terra-github-workflows/pull/117) has been merged, changed branch name back to `main`.

**Comment by ichengchang** [comment]
jenkins retest

**Pull request merged**